nexus/src/authz/context.rs

```diff
@@ -77,8 +81,29 @@ impl Context {
         resource: Resource,
     ) -> Result<(), Error>
     where
-        Resource: AuthorizedResource + Clone,
+        Resource: AuthorizedResource + oso::PolarClass + Clone,
     {
+        // If we're given a resource whose PolarClass was never registered with
+        // Oso, then the call to `is_allowed()` below will always return false
+        // (indicating that the actor does not have permissions).  That will
+        // cause this function to return an authz failure error (401, 403, or
+        // 404, depending on the context).  This is never what we intend.
+        // What's likely happened is that somebody forgot to register the class
+        // with Oso.  This failure mode is very hard to debug because the Rust
+        // code generates a valid Polar snippet and there's a working PolarClass
+        // impl -- it's just that neither was ever given to Oso.  Make this
+        // failure mode more debuggable by reporting a 500 with a clear error
+        // message.  After all, this is a bug.  (We could panic, since it's more
+        // of a programmer error than an operational error.  But unlike most
+        // programmer errors, the nature of the problem and the blast radius are
+        // well understood, so we may as well avoid crashing.)
+        let class_name = &Resource::get_polar_class().name;
```

Comment:
I think doing the check here on `name` is fine for our use case, because we don't ever expect to reuse class names. But I think technically it's the `type_id` that's the true unique identifier for a Polar class.

Reply:
That's interesting.  I guess if two things share a `type_id`, they almost certainly share a `name`.  But two things could share a name but have different types.  I assumed the class name had to be unique in order for Oso to map it to the corresponding actor or resource in Polar.  But I guess in other languages it's possible to use classes that have never been registered and aren't explicitly in the Polar file (because, say, they're subclasses of things that _are_ registered and named in the Polar file), and those wouldn't have to have unique names.

As you said, in our case, I think we expect them to be unique so I think I'd like to continue enforcing that.